pages/mdbook.yml

```diff
@@ -0,0 +1,60 @@
+# Sample workflow for building and deploying a mdbook site to GitHub Pages
+#
+# To get started with mdbook see: https://rust-lang.github.io/mdBook/index.html
+#
+name: Deploy mdbook site to Pages
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: [$default-branch]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v3
+
+      # Runs a single command using the runners shell
+      - name: install mdbook
+        run: |
+          curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf -y | sh 
+          rustup update
+          cargo install mdbook
```

Comment:
@Decodetalkers Is there an expected way we can trace this back to a given commit that published the `mdbook` executable? For example the way we do `hugo` is download the [GitHub Release](https://github.com/actions/starter-workflows/blob/8a95669da90bae4390caa6feb69cd30f70cffab3/pages/hugo.yml#L37). It could be nice to allow specifying a version as an input at least, and do `cargo install mdbook@{VERSION}`. That way the version of mdbook doesn't change between each page build.

Reply:
> 

Yes, cargo can install a pined version target, but which version you think is better?